client/dashboard/sites/settings-transfer-site/confirm-new-owner-form.tsx

```diff
@@ -0,0 +1,83 @@
+import { DataForm } from '@automattic/dataviews';
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	Button,
+} from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { sprintf, __ } from '@wordpress/i18n';
+import { useState } from 'react';
+import type { Field, SimpleFormField } from '@automattic/dataviews';
+
+type ConfirmNewOwnerFormData = {
+	email: string;
+};
+
+const fields: Field< ConfirmNewOwnerFormData >[] = [
+	{
+		id: 'email',
+		label: __( 'Email' ),
+		type: 'text' as const,
+	},
+];
+
+const form = {
+	type: 'regular' as const,
+	fields: [ { id: 'email' } as SimpleFormField ],
```

Comment:
Does it make sense to use `type=email`? We aren't giving any feedback to the user at the moment.

Reply:
Unfortunately, it doesn't support email type. See https://github.com/WordPress/gutenberg/blob/d7e0c5d872ba82d851c696cc4c67ea84368a5224/packages/dataviews/src/field-types/index.tsx#L15

Reply:
@oandregal Thoughts on adding email as a type? (re: https://github.com/WordPress/gutenberg/pull/64267)

Reply:
Added by [62ba76a](https://github.com/Automattic/wp-calypso/pull/103543/commits/62ba76a49a05355cb23cc39a2c25ad3d0c825607)

Reply:
Please wait a minute.... the existing logic allows typing username (not just email) and it will show you matching user 🥲 

<img width="746" alt="image" src="https://github.com/user-attachments/assets/6b5dcc13-e9cb-49fb-81a4-7d85ac8b6bd0" />

Reply:
But it's weird that the field is email, and the description also mentioned you have to enter the new email.

We can still show the matching users and enable the continue with validation 🙂

Reply:
Proposed https://github.com/Automattic/wp-calypso/pull/103579